src/garage/torch/algos/vpg.py

```diff
@@ -0,0 +1,286 @@
+
+        self._check_entropy_configuration(entropy_method, center_adv,
+                                          stop_entropy_gradient,
+                                          policy_ent_coeff)
+
+        self.reward_mean = collections.deque(maxlen=100)
+
+        self.optimizer = optimizer(
+            self.policy.get_module().parameters(), lr=self.policy_lr)
+
+    def _check_entropy_configuration(self, entropy_method, center_adv,
+                                     stop_entropy_gradient, policy_ent_coeff):
+        if entropy_method not in ('max', 'regularized', 'no_entropy'):
+            raise ValueError('Invalid entropy_method')
+
+        if entropy_method == 'max':
+            if center_adv:
+                raise ValueError('center_adv should be False when '
+                                 'entropy_method is max')
+            if not stop_entropy_gradient:
+                raise ValueError('stop_gradient should be True when '
+                                 'entropy_method is max')
+        if entropy_method == 'no_entropy':
+            if policy_ent_coeff != 0.0:
+                raise ValueError('policy_ent_coeff should be zero '
+                                 'when there is no entropy method')
+
+    def train_once(self, itr, paths):
+        """Perform one step of policy optimization."""
+        self._optimize_policy(itr, paths)
+        self._optimize_baseline(itr, paths)
+
+        return np.mean([sum(path['rewards']) for path in paths])
+
+    def _optimize_policy(self, itr, paths):
+        loss = [self._get_policy_loss(path) for path in paths]
+
+        self.optimizer.zero_grad()
+        torch.cat(loss).mean().backward()
+        self.optimizer.step()
+
+    def _get_policy_loss(self, path):
+        rewards = torch.Tensor(path['rewards'])
+        obs = torch.Tensor(path['observations'])
+
+        policy_entropy = self._get_policy_entropy(obs)
+
+        if self.maximum_entropy:
+            rewards = [
+                reward + self.policy_ent_coeff * policy_entropy
+                for reward in rewards
+            ]
+
+        baselines = self._get_baselines(path)
+        path['baselines'] = baselines.detach().numpy()
+        advantages = self._compute_advantages(baselines, rewards)
+
+        if self.center_adv:
+            advantages = F.batch_norm(
+                advantages.reshape((-1, 1)),
+                advantages.mean().unsqueeze(0),
+                advantages.var().unsqueeze(0),
+                eps=self.eps)
+            advantages = advantages.squeeze()
+
+        if self.positive_adv:
+            advantages -= advantages.min()
+
+        log_likelihood = self.policy.get_log_likelihood(
+            torch.Tensor(path['observations']), torch.Tensor(path['actions']))
+
+        grad = (log_likelihood * advantages).sum()
+
+        if self.entropy_regularzied:
+            grad += self.policy_ent_coeff * policy_entropy
+
+        return -grad.unsqueeze(0)  # Maximize
```

Comment:
I think you want to minimize this?

Reply:
I think I need to maximize the value and `Adam` optimizer try to minimize the value (loss).
This is why I put the "-" on the "grad" value.

Am I understanding correctly?

Reply:
Yes you are right. I think you can just remove the comment since it is a bit confusing.

Reply:
fixed - delete comment